lib/util/common.ts

```diff
@@ -59,3 +61,23 @@ export function detectPlatform(
 
   return null;
 }
+
+export function parseJsonWithFallback(content: string): any {
+  let parsedJson: any;
+
+  try {
+    parsedJson = JSON.parse(content);
+  } catch (err) {
+    try {
+      parsedJson = JSON5.parse(content);
+      logger.warn(
+        'JSON5.parse was used to parse the JSON data. Please check your json file'
+      );
```

Comment:
```suggestion
export function parseJsonWithFallback(content: string, context: string): any {
  let parsedJson: any;

  try {
    parsedJson = JSON.parse(content);
  } catch (err) {
    try {
      parsedJson = JSON5.parse(content);
      logger.warn(
        { context },
        'File contents are invalid JSON but parse using JSON5. Support for this will be removed in a future release so please change to a support .json5 file name or ensure correct JSON syntax.'      );
```

Reply:
Let's log the context (e.g. file name, or what was parsed) to help the user. Will require updating of all callers